### ISSUE TITLE
static build for 2.0.0-rc5

### ISSUE THREAD

**Issue by xx24678** [unicorn-engine/unicorn#1508]
How to do static build? I did the following:
```
UNICORN_VERSION=2.0.0-rc5
mkdir build && cd build
cmake \
    -DCMAKE_INSTALL_PREFIX="../../$UNICORN_VERSION" \
    -DUNICORN_BUILD_SHARED=OFF \
    -DCMAKE_BUILD_TYPE=Release \
    ..

make install -j8
```
This command ended up with only a 40KB libunicorn.a file in 2.0.0-rc5/lib directory, which only contains `cpu.c.o  uc.c.o  vl.c.o`. Am I doing something wrong here?

**Comment by wtdcode** [comment]
That’s intended. You have to link against all static libraries including libsoftmmu-* and libunicorn-common.

**Comment by xx24678** [comment]
any idea how to generate all those static libraries? My build with the above command only generates libunicorn.a.

**Comment by wtdcode** [comment]
> any idea how to generate all those static libraries? My build with the above command only generates libunicorn.a.

Those static libraries are also generated in the same directory.

**Comment by xx24678** [comment]
I see. They are generated in the build directory, but not copied to install direction.

**Comment by wtdcode** [comment]
> I see. They are generated in the build directory, but not copied to install direction.

Sorry I didn't see you are doing an installation.

Yes, you are right. It's due to cmake limitation and we don't have a cross-platform solution to merge static libraries (at least I don't know).

**Comment by wtdcode** [comment]
Fixed in 82559cb9a75c1bf84b04ba740cbf07641859aed4
